src/utils.jl

```diff
@@ -406,7 +406,10 @@ end
 `combinearg` is the inverse of [`splitarg`](@ref).
 """
 function combinearg(arg_name, arg_type, is_splat, default)
-    a = arg_name===nothing ? :(::$arg_type) : :($arg_name::$arg_type)
+    @assert arg_name !== nothing || arg_type !== nothing
+    a = arg_name===nothing ? :(::$arg_type) :
+        arg_type==:Any ? arg_name :
```

Comment:
```suggestion
        arg_type==:Any && is_splat ? arg_name :   # see #177 and julia#43625
```
I hate having a special case here, but if we're going to have one, might as well make it as specific as possible to reduce the chance that someone will bump into it. It's possible to have a `@outer @inner f(...) = ` situation where `@outer` expects all arguments to be typed, and `@inner` uses `combinearg` to achieve that.